tests/phpunit/StreamWrapper/IncludeInterceptorTest.php

```diff
@@ -223,4 +230,188 @@ public function test_passthrough_dir_methods_pass(): void
+
+        IncludeInterceptor::intercept(self::$files[1], self::$files[2]);
+        IncludeInterceptor::enable();
+
+        try {
+            include 'file-does-not-exist.txt';
+        } catch (Warning $e) {
+            $after = include self::$files[1];
+
+            $this->assertSame($after, $expected);
+
+            return;
+        }
+
+        $this->fail('Badly set up test, exception was not thrown');
+    }
+
+    public function test_it_re_enables_interceptor_after_directory_not_found_with_opendir(): void
+    {
+        $expected = include self::$files[2];
+
+        IncludeInterceptor::intercept(self::$files[1], self::$files[2]);
+        IncludeInterceptor::enable();
+
+        try {
+            opendir('dir-does-not-exist');
+        } catch (Warning $e) {
+            $after = include self::$files[1];
+
+            $this->assertSame($after, $expected);
+
+            return;
+        }
+
+        $this->fail('Badly set up test, exception was not thrown');
+    }
+
+    public function test_it_re_enables_interceptor_after_directory_already_exist_with_mkdir(): void
+    {
+        $expected = include self::$files[2];
+
+        IncludeInterceptor::intercept(self::$files[1], self::$files[2]);
+        IncludeInterceptor::enable();
+
+        try {
+            mkdir(__DIR__);
+        } catch (Warning $e) {
+            $after = include self::$files[1];
+
+            $this->assertSame($after, $expected);
+
+            return;
+        }
+
+        $this->fail('Badly set up test, exception was not thrown');
+    }
+
+    public function test_it_re_enables_interceptor_after_file_not_found_with_rename(): void
+    {
+        $expected = include self::$files[2];
+
+        IncludeInterceptor::intercept(self::$files[1], self::$files[2]);
+        IncludeInterceptor::enable();
+
+        try {
+            rename('file-does-not-exist.txt', 'something-else.txt');
+        } catch (Warning $e) {
+            $after = include self::$files[1];
+
+            $this->assertSame($after, $expected);
+
+            return;
+        }
+
+        $this->fail('Badly set up test, exception was not thrown');
+    }
+
+    public function test_it_re_enables_interceptor_after_directory_not_found_with_rmdir(): void
+    {
+        $expected = include self::$files[2];
+
+        IncludeInterceptor::intercept(self::$files[1], self::$files[2]);
+        IncludeInterceptor::enable();
+
+        try {
+            rmdir('dir-does-not-exist');
+        } catch (Warning $e) {
+            $after = include self::$files[1];
+
+            $this->assertSame($after, $expected);
+
+            return;
+        }
+
+        $this->fail('Badly set up test, exception was not thrown');
+    }
+
+    public function test_it_re_enables_interceptor_after_file_not_found_with_stream_metadata(): void
+    {
+        if ('\\' === \DIRECTORY_SEPARATOR) {
+            $this->markTestSkipped('This test cannot be run on Windows.');
```

Comment:
If this is correct, the more canonical way to express this would be with the `@requires` annotation, e.g. `@requires osfamily Windows`. However, I suspect it is probably not correct, as most file operations are valid on Windows/NTFS, too. Not really sure what "metadata" is supposed to be testing, though.

Reply:
i've copied that bit of code someplace else in the codebase. happy to ear what @BackEndTea  thinks of this, sure

Reply:
`stream_metadata` is called by other system functions https://www.php.net/manual/en/streamwrapper.stream-metadata.php. but it seems to be implemented differently on Windows and doesn't emit warnings or errors in invalid situations

Reply:
Seems like [the recommended way](https://phpunit.readthedocs.io/en/8.4/incomplete-and-skipped-tests.html#skipping-tests-using-requires) is using `@requires OSFAMILY Windows` as @Bilge said. Let's use it

Reply:
Might be done in a separate PR to fix it elsewhere then

Reply:
Done in https://github.com/infection/infection/pull/856